internal/datastore/postgres/xid8.go

```diff
@@ -139,7 +139,7 @@ func (c Uint64Codec) DecodeDatabaseSQLValue(m *pgtype.Map, oid uint32, format in
 	if err != nil {
 		return nil, err
 	}
-	return int64(n), nil
+	return n, nil
```

Comment:
This is the only change that makes me really uneasy. I'm not sure why this cast was here in the first place. Lint isn't complaining; we'll see whether there are test issues.